File: C#/atividades/atividade5/FormasGeometricas/FormasGeometricas/Models/Circulo.cs
namespace FormasGeometricas.FormaGenerica;

internal class Circulo : FormaGenerica, ICalculoArea
{
    public float Raio { get; set; }

    public Circulo(string nome, float raio) : base(nome)
    {
        Raio = raio;
    }

    public override void ExibirInfo()
    {
        base.ExibirInfo();
        CalcularArea();
        CalcularPerimetro();
    }

    public void CalcularArea()
    {
        Console.WriteLine($"Área do {Nome}: {3.14 * (Raio * Raio)}");
    }
    public void CalcularPerimetro()
    {
        Console.WriteLine($"Perimetro do {Nome}: {2 * 3.14 * Raio}");
    }
}

File: JAVA/ORIENTACAO-A-OBJETO/ATIVIDADES/orientacao-a-objeto/video2/atividade  4 ok/FormasGeometricas/src/Circulo.java
public class Circulo extends FormaGeometrica implements ICalculavel {
    private double Pi = 3.14;
    private double Raio;

    public Circulo(String cor, String nome, double raio) {
        super(cor, nome);

        this.Raio = raio;
    }


    @Override
    public void calcularArea() {
        System.out.println("Área do círculo: " + this.Pi * this.Raio * this.Raio);
    }

    @Override
    public void calcularPerimetro() {
        System.out.println("Perímetro do círculo: " + (2 * this.Pi + this.Raio));
    }

}
